### PR TITLE
Mark `DefineOpaqueTypes::Yes` use-sites uncovered by tests as FIXMEs

### PR DESCRIPTION
**Pull request by lqd**
I noticed #116652 earlier, and this PR marks all the `DefineOpaqueTypes::No` use-sites that didn't make any test fail when flipped to `DefineOpaqueTypes::Yes` (each into their dedicated commit as requested in that issue).

r? @compiler-errors

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
duplicate of https://github.com/rust-lang/rust/pull/117348 ? :sweat_smile:

File: compiler/rustc_hir_analysis/src/coherence/builtin.rs [hunk 1]
```diff
@@ -165,6 +165,7 @@ fn visit_implementation_of_dispatch_from_dyn(tcx: TyCtxt<'_>, impl_did: LocalDef
     use rustc_type_ir::TyKind::*;
     match (source.kind(), target.kind()) {
         (&Ref(r_a, _, mutbl_a), Ref(r_b, _, mutbl_b))
+            // FIXME(DefineOpaqueTypes): no test exercizes using `DefineOpaqueTypes::Yes` below.
```
Comment (compiler-errors) [inline compiler/rustc_hir_analysis/src/coherence/builtin.rs:168]:
```suggestion
            // FIXME(DefineOpaqueTypes): no test exercises using `DefineOpaqueTypes::Yes` below.
```

**Review by compiler-errors — COMMENTED** [review]
Hm... I actually wonder if we may just want to track these usage sites as issues, rather than FIXMEs in the code.

Ideally, I would like to have interested individuals (or maybe just myself) study the way that the code is being used and create tests from that (or conclude that tests cannot be created and switch them to `DefineOpaqueTypes::Yes`).

Also, each usage site alone is a bit too granular. Except for candidate assembly and selection in the new trait solver, we probably want to group (e.g.) all of the method probe usages together, and all of the coercion usages together, etc.

**Comment by lqd** [comment]
> duplicate of #117348 ? 😅

I saw that after starting and it's more complementary: it flipped uses to `DefineOpaqueTypes::Yes` which I didn't do (and I don't believe was expected, if I understood the steps in #116652 correctly), and didn't mark the places that were not covered by tests which I did do. 

Some of the new `DefineOpaqueTypes::Yes` in that PR will land, while all the FIXMEs for the others that probably won't land are in this PR.

File: compiler/rustc_hir_analysis/src/coherence/builtin.rs (hunk 1 above)
Comment (lqd) [reply compiler/rustc_hir_analysis/src/coherence/builtin.rs:168]: TIL

**Comment by compiler-errors** [comment]
And yeah, this work is kind-of underway in #117348.

I actually slightly regret the approach suggested #116652. Seems a bit too much like busy work -- creating 36 commits that all just add a single `FIXME` to a file seems... kinda pointless?

Ideally we'd be working incrementally towards our goal of actually getting rid of `DefineOpaqueTypes::No`, and while this is a step in the right direction, it also seems a bit like churn.

**Comment by lqd** [comment]
Ok then let's close this.

**Pull request closed without merging**